Ported RailPowerLimit

DeadlyMC opened the pull request:
Added patch files and enabled carpet rule

DeadlyMC committed:
RailPowerLimit

Added patch files and enabled carpet rule

gnembon commented on patches/net/minecraft/block/BlockRailPowered.java.patch:
```diff
@@ -0,0 +1,19 @@
+-        if (p_176566_5_ >= 8)
++        // [CM] Changed if statement values -- RailPowerLimit
++        if (p_176566_5_ >= CarpetSettings.getInt("railPowerLimit"))
+         {
```
can you like with push limit, add numAccelerate() to the rule? with a large rail system this may start to lag

gnembon commented on src/main/java/carpet/CarpetSettings.java:
```diff
@@ -236,8 +236,8 @@ private static void set_defaults()
                                 .choices("12","10 12 14 100").setNotStrict().numAccelerate(),
-  //!rule("railPowerLimit",        "creative", "Customizable powered rail power range")
-  //                              .choices("9","9 15 30").setNotStrict(),
+  rule("railPowerLimit",        "creative", "Customizable powered rail power range")
```
you would also need to add public static int n_railPowerLimit = 8;

DeadlyMC committed:
Using numAccelerate()

Old system might have caused lag with a large rail system, using numAccelerate() should fix it.

DeadlyMC commented:
All done :)

gnembon committed:
adjusted power limit
